internal/transport/proxy_test.go

```diff
@@ -84,7 +85,24 @@ func (p *proxyServer) run() {
 		return
 	}
 	resp := http.Response{StatusCode: http.StatusOK, Proto: "HTTP/1.0"}
-	resp.Write(p.in)
+	var buf bytes.Buffer
+	resp.Write(&buf)
+	// Batch the first message from the server with the http connect
+	// response. This is done to test the cases in which the grpc client has
+	// the response to the connect request and proxied packets from the
+	// destination server when it reads the transport.
+	out.SetReadDeadline(time.Now().Add(20 * time.Millisecond))
+	b := make([]byte, 50)
+	bytesRead, err := out.Read(b)
+	// The read is expected to fail with deadline exceeded if the server doesn't
+	// have a message to send.
+	if err != nil {
+		p.t.Logf("Got error while reading message from server: %v", err)
+	}
```

Comment:
Is there another way to test this functionality that we can use instead of setting a magic deadline?

Maybe a more directed unit test of the `doHTTPConnectHandshake` function?

Or parameterize the proxyServer so it knows to do this step / skip it, and set a longer deadline and fail if it errors?

Reply:
Parameterize proxy server to wait for the server hello.